### PR TITLE
feat: add C implementation `stats/base/dists/rayleigh/pdf`

### LINKED ISSUE #3807: [RFC]: Add C implementation for `@stdlib/stats/base/dists/rayleigh/pdf`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/rayleigh/pdf`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/rayleigh/pdf" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/rayleigh/pdf/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Vinit-Pandit**
Resolves #3807.

## Description

> What is the purpose of this pull request?

This pull request:

-   add c implementation `stats/base/dists/rayleigh/pdf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3807 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+* Evaluates the probability density function (PDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+*
+* @param x    input value
+* @param a    scale parameter
+* @return     evaluated PDF
+*
+* @example
+* double y = stdlib_base_arcsine_cdf( 3.0, 1.0 );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c]:
```suggestion
* double y = stdlib_base_rayleigh_cdf( 3.0, 1.0 );
```

**Comment by MeastroZI** [comment]
Let me have a review first :)

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+*/
+double stdlib_base_dists_rayleigh_pdf( const double x, const double sigma ) {
+	double s2;
+	double s2i;
+	if (
+		stdlib_base_is_nan( x ) ||
+		stdlib_base_is_nan( sigma ) ||
+		sigma < 0.0
+	) {
+		return 0.0/0.0; // NaN
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c]:
```suggestion
		return 0.0 / 0.0; // NaN
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#include "stdlib/math/base/special/exp.h"
+#include "stdlib/math/base/special/pow.h"
+#include "stdlib/constants/float64/pinf.h"
+
+
+/**
+* Evaluates the probability density function (PDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+*
+* @param x    input value
+* @param a    scale parameter
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c]:
```suggestion
* @param sigma    scale parameter
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Evaluates the probability density function (PDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+*
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/lib/native.js:30]:
```suggestion
*
* @private
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/include/stdlib/stats/base/dists/rayleigh/pdf.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Evaluates the probability distribution function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/include/stdlib/stats/base/dists/rayleigh/pdf.h]:
```suggestion
* Evaluates the probability density function (PDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/benchmark/c/benchmark.c [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	double elapsed;
+	double x[ 100 ];
+	double sigma[ 100 ];
+	double y;
+	double t;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = random_uniform( -100.0, 0.0 );
+		sigma[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS , STDLIB_CONSTANT_FLOAT64_EPS + 20 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/benchmark/c/benchmark.c]:
```suggestion
		sigma[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS , STDLIB_CONSTANT_FLOAT64_EPS + 20.0 );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/benchmark/c/benchmark.c [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+#include "stdlib/stats/base/dists/rayleigh/pdf.h"
+#include "stdlib/constants/float64/eps.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <time.h>
+#include <sys/time.h>
+
+#define NAME "rayleigh-cdf"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/benchmark/c/benchmark.c]:
```suggestion
#define NAME "rayleigh-pdf"
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/examples/c/example.c [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+static double random_uniform( const double min, const double max ) {
+	double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
+	return min + ( v*(max-min) );
+}
+
+int main( void ) {
+	double x;
+	double sigma;
+	double y;
+	int i;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/examples/c/example.c]:
```suggestion
	double sigma;
	double x;
	double y;
	int i;
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/README.md [hunk 9]
```diff
@@ -143,6 +143,101 @@ for ( i = 0; i < 10; i++ ) {
...
+static double random_uniform( const double min, const double max ) {
+    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
+    return min + ( v*(max-min) );
+}
+
+int main( void ) {
+    double x;
+    double sigma;
+    double y;
+    int i;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/README.md]:
```suggestion
    double sigma;
    double x;
    double y;
    int i;
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/rayleigh/pdf.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/exp.h"
+#include "stdlib/math/base/special/pow.h"
+#include "stdlib/constants/float64/pinf.h"
+
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c]:
```suggestion

```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+/**
+* Evaluates the probability density function (PDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+*
+* @param x    input value
+* @param sigma    scale parameter
+* @return     evaluated PDF
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/pdf/src/main.c]:
```suggestion
* @param x        input value
* @param sigma    scale parameter
* @return         evaluated PDF
```

**Review by Planeshifter — APPROVED** [review]
Thank you, @MeastroZI!

There were a few minor issues in the PR but those should now be resolved. Will merge shortly after CI has passed.

**Comment by Vinit-Pandit** [comment]
Thank You @Planeshifter , I will be more careful from next time for this kind of issues.

**Pull request merged**